code/model/SiteTree.php

```diff
@@ -2643,19 +2609,18 @@ public function setMenuTitle($value) {
 	}
 	
 	/**
-	 * A flag provides the user with additional data about the current page status,
-	 * for example a "removed from draft" status. Each page can have more than one status flag.
-	 * Returns a map of a unique key to a (localized) title for the flag.
-	 * The unique key can be reused as a CSS class.
-	 * Use the 'updateStatusFlags' extension point to customize the flags.
+	 * A flag provides the user with additional data about the current page status, for example a "removed from draft"
+	 * status. Each page can have more than one status flag. Returns a map of a unique key to a (localized) title for
+	 * the flag. The unique key can be reused as a CSS class. Use the 'updateStatusFlags' extension point to customize
+	 * the flags.
 	 * 
 	 * Example (simple): 
-	 * "deletedonlive" => "Deleted"
+	 *   "deletedonlive" => "Deleted"
 	 * 
 	 * Example (with optional title attribute): 
-	 * "deletedonlive" => array('text' => "Deleted", 'title' => 'This page has been deleted')
+	 *   "deletedonlive" => array('text' => "Deleted", 'title' => 'This page has been deleted')
 	 *
-	 * @param Boolean $cached
+	 * @param bool $cached Whether to serve the fields from cache; false regenerate them
```

Comment:
There are no scalar type hints (yet), so all scalar types are loose comparisons. The annotations are our way of hinting at strict, scalar types. We could leave the types or hint mixed. If we're being specific (this is `int`, this is `string etc.) then we should be as specific as possible.

The [PSR recommends](https://github.com/phpDocumentor/fig-standards/blob/master/proposed/phpdoc.md#details) being as detailed as possible.

The reason for all the detail is so that a static analyser can predict errors before runtime. If you don't annotate `null|bool` on an optional parameter then PHPStorm (or whatever static analyser you use) will warn you that you could be passing null to function that doesn't expect it.